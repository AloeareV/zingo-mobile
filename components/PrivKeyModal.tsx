/* eslint-disable react-native/no-inline-styles */
import React, {useState} from 'react';
import {View, ScrollView, SafeAreaView, Image, Text, Platform, TouchableOpacity} from 'react-native';
import Clipboard from '@react-native-community/clipboard';
import {ClickableText, FadeText} from './Components';
import Button from './Button';
import {useTheme} from '@react-navigation/native';
import Utils from '../app/utils';
import Toast from 'react-native-simple-toast';
import QRCode from 'react-native-qrcode-svg';

type PrivKeyModalProps = {
  closeModal: () => void;
  address: string;
  keyType: number;
  privKey: string;
};
const PrivKeyModal: React.FunctionComponent<PrivKeyModalProps> = ({address, keyType, privKey, closeModal}) => {
  const {colors} = useTheme();
  const fixedWidthFont = Platform.OS === 'android' ? 'monospace' : 'Courier';

  const keyTypeString = keyType === 0 ? 'Private' : 'Viewing';
  
  // 30 characters per line
  const numLines = (privKey.length / 30);
  const keyChunks = Utils.splitStringIntoChunks(privKey, numLines.toFixed(0));

  const [expandAddress, setExpandAddress] = useState(false);

  if (!privKey) {
    privKey = 'No Key';
  }

  const doCopy = () => {
    if (address) {
      Clipboard.setString(privKey);
      Toast.show('Copied key to Clipboard', Toast.LONG);
    }
  };

  return (
    <SafeAreaView
      style={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'stretch',
        height: '100%',
        backgroundColor: colors.background,
      }}>
      <View>
        <View style={{alignItems: 'center', backgroundColor: colors.card, paddingBottom: 25, paddingTop: 25}}>
          <Text style={{marginTop: 5, padding: 5, color: colors.text, fontSize: 28}}>{keyTypeString} Key</Text>
          <TouchableOpacity
            onPress={() => {
              setExpandAddress(true);
            }}>
            <FadeText style={{textAlign: 'center', marginLeft: 10, marginRight: 10}}>
              {expandAddress ? address : Utils.trimToSmall(address, 10)}
            </FadeText>
          </TouchableOpacity>
        </View>
        <View style={{display: 'flex', alignItems: 'center', marginTop: -25}}>
          <Image source={require('../assets/img/logobig-zingo.png')} style={{width: 50, height: 50, resizeMode: 'contain'}} />
        </View>
      </View>

      <ScrollView
        style={{maxHeight: '85%'}}
        contentContainerStyle={{
          flexDirection: 'column',
          alignItems: 'stretch',
          justifyContent: 'flex-start',
        }}>
        <View style={{display: 'flex', flexDirection: 'column', marginTop: 10, alignItems: 'center'}}>
          <View style={{padding: 10, backgroundColor: 'rgb(255, 255, 255)', marginTop: 15, marginBottom: 20}}>
            <QRCode value={privKey} size={225} ecl="L" />
          </View>
          <ClickableText style={{marginBottom: 5}} onPress={doCopy}>
            Tap To Copy
          </ClickableText>

          {keyChunks.map(c => (
            <FadeText
              key={c}
              style={{
                flexBasis: '100%',
                textAlign: 'center',
                fontFamily: fixedWidthFont,
                fontSize: 18,
                color: colors.text,
              }}>
              {c}
            </FadeText>
          ))}
        </View>
      </ScrollView>
      <View style={{flexGrow: 1, flexDirection: 'row', justifyContent: 'center', alignItems: 'center', margin: 10}}>
        <Button type="Secondary" title="Close" onPress={closeModal} />
      </View>
    </SafeAreaView>
  );
};

export default PrivKeyModal;
