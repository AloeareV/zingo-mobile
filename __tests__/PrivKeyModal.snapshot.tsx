/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import PrivKeyModal from '../components/PrivKeyModal';

jest.useFakeTimers()

// test suite
describe("Component PrivKeyModal - test", () => {
  //snapshot test
  test("PrivKeyModal - snapshot", () => {
    const privKeyModal = create(<PrivKeyModal address={''} keyType={0} privKey={''} closeModal={() => {}} />);
    expect(privKeyModal.toJSON()).toMatchSnapshot();
  });

});
