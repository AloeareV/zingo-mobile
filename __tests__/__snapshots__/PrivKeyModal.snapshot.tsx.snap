// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrivKeyModal - test PrivKeyModal - snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "paddingBottom": 25,
          "paddingTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontSize": 28,
            "marginTop": 5,
            "padding": 5,
          }
        }
      >
        Private
         Key
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "marginLeft": 10,
              "marginRight": 10,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginTop": -25,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/logobig-zingo.png",
          }
        }
        style={
          Object {
            "height": 50,
            "resizeMode": "contain",
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      Object {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "marginBottom": 20,
              "marginTop": 15,
              "padding": 10,
            }
          }
        >
          <mockQRCode />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "marginBottom": 5,
                "textDecorationLine": "underline",
              }
            }
          >
            Tap To Copy
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "100%",
              "fontFamily": "Courier",
              "fontSize": 18,
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Close
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
